### ISSUE TITLE
[RyuJIT/ARM32] Assertion failed 'varDsc->lvOnFrame'

### ISSUE THREAD

**Issue by sjsinju** [dotnet/coreclr#12734]
```
Assertion failed 'varDsc->lvOnFrame'
File: /opt/code/github/hqueue/coreclr/src/jit/codegenarmarch.cpp Line: 668
Image: /home/dotnet_jenkins/workspace/hqueue/Run_coreclr_tests_Checked_ryujit_nyi_all/runtime/tests/Tests/coreoverlay/corerun
```

```
JIT/jit64/hfa/main/testA/hfa_sd2A_d/hfa_sd2A_d.sh
JIT/jit64/hfa/main/testC/hfa_sd2C_d/hfa_sd2C_d.sh
JIT/jit64/hfa/main/testG/hfa_sd2G_d/hfa_sd2G_d.sh
```

**Comment by sjsinju** [comment]
@BruceForstall Please assign this to me.

**Comment by hseok-oh** [comment]
@sjsinju This issue may be related with #12708.

**Comment by sjsinju** [comment]
@hseok-oh Yes. You're right. Thank you.

**Comment by sjsinju** [comment]
Fixed with #12708
